### PR TITLE
Descriptives

### PR DESCRIPTION
**Pull request by NicholasDanks**
This PR attends to Issue #54 a new feature.
summary(seminr_model) now return a descripts_stats list object containing 4 matrices:
item_descriptives
item_correlations
construct_descriptives
construct_descriptives

### CODE REVIEW

File: R/library.R [hunk 1]
```diff
@@ -257,3 +257,44 @@ measure_interaction <- function(intxn) {
     composite(intxn$name, colnames(intxn$data),weights = mode_A)
   }
 }
+
+kurt <- function(x, na.rm = FALSE) {
```
Comment (soumyaray) [inline R/library.R]: Few questions:
1. did you write these functions yourself or did you get them from somewhere?
2. do we need to handle all forms of objects (vector, matrix, data.frame etc.) or just vectors?

File: R/library.R [hunk 2]
```diff
@@ -257,3 +257,44 @@ measure_interaction <- function(intxn) {
...
+
+get_desc <- function(data, na.rm = na.rm) {
+  Mean <- apply(data, 2, mean)
+  Std.Dev. <- apply(data, 2, sd)
+  Kurtosis <- kurt(data, na.rm = na.rm)
+  Min <- apply(data, 2, min)
+  Max <- apply(data, 2, max)
+  Median <- apply(data, 2, median)
+  Skewness <- skew(data, na.rm = na.rm)
+  Missing <- apply(data, 2, function(x) sum(stats::complete.cases(x)==FALSE))
```
Comment (soumyaray) [inline R/library.R]: is `stats` base R?

File: R/report_descriptives.R [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# A funtion to take a seminr model and return item and construct descriptives
+get_descriptives <- function(seminr_model, na.rm = TRUE) {
+  #items
+  item_descriptives <- get_desc(seminr_model$data, na.rm = na.rm)
+  item_correlations <- cor(seminr_model$data)
+  #constructs
+  construct_descriptives <- get_desc(seminr_model$construct_scores, na.rm = na.rm)
+  construct_correlations <- cor(seminr_model$construct_scores)
+  class(item_descriptives) <- append(class(item_descriptives),"table_output")
```
Comment (soumyaray) [inline R/report_descriptives.R]: could you make it a practice to put a space after all commas?

File: R/report_descriptives.R [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# A funtion to take a seminr model and return item and construct descriptives
+get_descriptives <- function(seminr_model, na.rm = TRUE) {
```
Comment (soumyaray) [inline R/report_descriptives.R]: no need for prefix `get_` in function names. i think it's obvious that `descriptives()` would return descriptives.

File: vignettes/SEMinR.Rmd [hunk 5]
```diff
@@ -405,6 +405,7 @@ summary(mobi_pls)
     + `model_summary$composite_scores` reports the construct scores of composites
     + `model_summary$vif_items` reports the Variance Inflation Factor (VIF) for the measurement model
     + `model_summary$vif_antecedents` report the Variance Inflation Factor (VIF) for the structural model
+    + `model_summary$descriptive_stats` reports the descriptive statistics and correlations for both items and constructs
```
Comment (soumyaray) [inline vignettes/SEMinR.Rmd]: I would suggest: `descriptive_stats` —> `descriptives` or `statistics` or `descriptive_statistics`

File: R/report_descriptives.R [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  item_descriptives <- get_desc(seminr_model$data, na.rm = na.rm)
+  item_correlations <- cor(seminr_model$data)
+  #constructs
+  construct_descriptives <- get_desc(seminr_model$construct_scores, na.rm = na.rm)
+  construct_correlations <- cor(seminr_model$construct_scores)
+  class(item_descriptives) <- append(class(item_descriptives),"table_output")
+  class(item_correlations) <- append(class(item_correlations),"table_output")
+  class(construct_descriptives) <- append(class(construct_descriptives),"table_output")
+  class(construct_correlations) <- append(class(construct_correlations),"table_output")
+  return(list(item_descriptives = item_descriptives,
```
Comment (soumyaray) [inline R/report_descriptives.R]: should we make a class out of the return object? see `summary.seminr_model()`

File: vignettes/SEMinR.Rmd [hunk 7]
```diff
@@ -479,6 +480,23 @@ confidence_interval(boot_seminr_model = boot_seminr_model,
...
+The `summary(seminr_model)` function will return four matrices:
+`model_summary <- summary(seminr_model)` returns an object of class `summary.seminr_model` which contains the following four descriptive statistics matrices:
+    + `model_summary$descriptive_stats$item_descriptives` reports the descriptive statistics for items
+    + `model_summary$descriptive_stats$item_correlations` reports the correlation matrix for items
+    + `model_summary$descriptive_stats$construct_descriptives` reports the descriptive statistics for constructs
+    + `model_summary$descriptive_stats$construct_correlations` reports the correlation matrix for constructs
+
+```{r}
+model_summary <- summary(mobi_pls)
+model_summary$descriptive_stats$item_descriptives
```
Comment (soumyaray) [inline vignettes/SEMinR.Rmd]: What do you think if try to DRY the names a bit:
`model_summary$descriptives$correlations$items`
`model_summary$descriptives$descriptives$constructs`

**Comment by soumyaray** [comment]
apart from my comments, i notice travis ci tests are failing?

File: R/library.R (hunk 2 above)
Comment (NicholasDanks) [reply R/library.R]: Sure is

File: R/library.R (hunk 1 above)
Comment (NicholasDanks) [reply R/library.R]: 1. I altered them from elsewhere. 
2. Nope, so I have altered the code to just use apply

File: R/report_descriptives.R (hunk 3 above)
Comment (NicholasDanks) [reply R/report_descriptives.R]: I will do. I have ran through the code and I believe removed this problem.

File: R/report_descriptives.R (hunk 4 above)
Comment (NicholasDanks) [reply R/report_descriptives.R]: Done.

File: vignettes/SEMinR.Rmd (hunk 5 above)
Comment (NicholasDanks) [reply vignettes/SEMinR.Rmd]: Changed

**Comment by NicholasDanks** [comment]
> apart from my comments, i notice travis ci tests are failing?
A: Yes, the container that tries to install R development release is currently failing - not the code.
*The command "Rscript -e 'deps <- remotes::dev_package_deps(dependencies = NA);remotes::install_deps(dependencies = TRUE);if (!all(deps$package %in% installed.packages())) { message("missing: ", paste(setdiff(deps$package, installed.packages()), collapse=", ")); q(status = 1, save = "no")}'" failed and exited with 1 during .*

File: R/report_descriptives.R (hunk 6 above)
Comment (NicholasDanks) [reply R/report_descriptives.R]: I want users to be able to print the tables directly in a neat format and so I am classing the tables and not the return object.

File: R/library.R (hunk 1 above)
Comment (soumyaray) [reply R/library.R]: ok, in future just run by me once where we are taking code from.  we should know what license applies to that code.

**Pull request merged**